app/usecases/leaderboards.py

```diff
@@ -34,12 +35,14 @@ async def fetch_beatmap_leaderboard(
 
     int_mods_filter = int(mods_filter) if mods_filter else None
 
+    sort_column = mode.sort if not vanilla_pp_leaderboards else "pp"
```

Comment:
this is making an assumption that all other modes will use pp, which is true right now but not necessarily always the case -- can we check the vanilla_pp_leaderboardss for only mode=0